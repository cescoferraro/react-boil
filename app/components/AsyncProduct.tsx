import * as React from "react"
import { asyncComponent } from 'react-async-component'

export default asyncComponent({
    resolve: () => Promise.resolve().then(() => require("./Product.tsx"))
});
