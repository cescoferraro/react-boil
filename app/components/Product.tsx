import * as React from "react"
declare var System: any
import * as CSS from "../css/teste.css"

export default () => (
    <div className={CSS.test}>
        <h2>I am on a separate Bundle!</h2>
    </div>
)
