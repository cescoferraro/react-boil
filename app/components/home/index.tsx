import * as React from "react"
import * as CSS from "./css/home.css"
const test = require("../../../README.org")

export const HomeComponent = ({ }) => {
    return <div className={CSS.container} dangerouslySetInnerHTML={{ __html: test }} />
}
