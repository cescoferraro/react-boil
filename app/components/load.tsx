import universal from "react-universal-component"

export const UniversalComponent = universal(() => import("./product"), {
    resolve: () => require.resolveWeak("./product"),
    chunkName: "Universla",
    onError: (erros) => {
        console.log(erros)
    }
})
