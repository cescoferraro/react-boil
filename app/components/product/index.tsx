import universal from "react-universal-component"
declare var System: any

export const UniversalComponent = universal(
    () => System.import(/* webpackChunkName: 'Product' */ "./Product"),
    {
        resolve: () => require.resolveWeak("./Product"),
        chunkName: "Product"
    }
)
