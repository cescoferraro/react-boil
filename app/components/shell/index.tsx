import { ListSubheader as Subheader } from 'material-ui/List'
import * as cs from "classnames"
import * as React from "react"
import * as CSS from "./shell.css"
import Drawer from "material-ui/Drawer"
import Menu, { MenuItem } from 'material-ui/Menu';
const BoilLogo = require("../../../shared/images/boil.svg")
const Cancel = require("../../../shared/images/cancel.svg")
const Boy = require("../../../shared/images/boy.svg")
const Girl = require("../../../shared/images/girl.svg")
import ReduxToastr from "react-redux-toastr"
import { withStyles, createStyleSheet } from 'material-ui/styles';
import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import Typography from 'material-ui/Typography';
import Button from 'material-ui/Button';
import IconButton from 'material-ui/IconButton';
import MenuIcon from 'material-ui-icons/Menu';

const styleSheet = createStyleSheet({
    root: {
        width: '100%',
    },
    flex: {
        flex: 1,
    },
});



export const Shell = withStyles(styleSheet)((props) => {
    const closeOnClick = (open) => { props.DRAWER_ACTION(open) }
    const boilLogo = (
        <BoilLogo
            onClick={props.DRAWER_TOGGLE_ACTION}
            className={CSS.button}
        />
    )
    const goBack = () => {
        props.GO_HOME_ACTION()
        props.DRAWER_ACTION(false)
    }
    const goUser = (id) => () => {
        props.USER_ACTION(id)
        props.DRAWER_ACTION(false)
    }
    return (
        <div>
            <div className={props.classes.root}>
                <AppBar position="static">
                    <Toolbar>
                        <IconButton color="contrast" aria-label="Menu">
                            <MenuIcon />
                        </IconButton>
                        <Typography
                            onClick={goBack}
                            type="title" color="inherit" className={props.classes.flex}>
                            React-boil
			</Typography>

                        <BoilLogo
                            onClick={props.DRAWER_TOGGLE_ACTION}
                            className={CSS.button}
                        />
                    </Toolbar>
                </AppBar>
            </div>
            <div className={CSS.container}>
                {props.children}
            </div>
            <ReduxToastr
                timeOut={4000}
                newestOnTop={false}
                preventDuplicates={true}
                position="top-left"
                transitionIn="fadeIn"
                transitionOut="fadeOut"
                progressBar={true}
            />
            <Drawer
                docked={false}
                open={props.drawer}
            >
                <div className={cs(CSS.flex, CSS.main)}>
                    <BoilLogo
                        onClick={goBack}
                        className={cs(CSS.svg)}
                    />
                </div>
                <Subheader>React-boil</Subheader>
                <MenuItem
                    onClick={goUser(432)}
                >
                    User 123
                </MenuItem>
                <MenuItem
                    onClick={goUser(324)}
                >
                    User 456
                </MenuItem>
                <MenuItem
                    role="ksdjnf"
                    onClick={props.DRAWER_TOGGLE_ACTION}
                >
                    Close
                </MenuItem>
            </Drawer>
        </div>
    )
})
