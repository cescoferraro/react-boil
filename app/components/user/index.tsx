import universal from "react-universal-component"
declare var System: any

export const AsyncUser = universal(
    () => System.import(/* webpackChunkName: 'user' */ "./user"),
    {
        resolve: () => require.resolveWeak("./user"),
        chunkName: "user"
    }
)
