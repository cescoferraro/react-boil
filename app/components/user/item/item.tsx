import * as React from "react"

export const Item = ({ CSS, icon, title, value }) => {
    return (
        <div className={CSS.item}>
            <div className={CSS.itemIcon}>
                {icon}
            </div>
            <div className={CSS.itemText}>
                <a className={CSS.key}> {title} </a>
                <a className={CSS.value}> {value} </a>
            </div>
        </div>
    )
}
