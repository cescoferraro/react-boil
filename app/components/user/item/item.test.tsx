import * as React from "react"
import { Item } from "./item"
import { MUIProvider } from "../../../../shared/components/MUIProvider"
import GpsFixed from "material-ui-icons/GpsFixed.js"

test("Render an Item", () => {
    const icon = (<GpsFixed style={{ fill: "grey" }} />)
    const component = mount(
        <MUIProvider>
            <Item
                title="Email"
                value={"teste@example.com"}
                icon={icon}
            />
        </MUIProvider>
    )
    expect(component).toMatchSnapshot()
})
