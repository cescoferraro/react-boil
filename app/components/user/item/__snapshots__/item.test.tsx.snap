// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render an Item 1`] = `
<Component>
  <MuiThemeProvider
    muiTheme={
      Object {
        "appBar": Object {
          "color": "#303F9F",
          "height": 64,
          "padding": 24,
          "textColor": "#ffffff",
          "titleFontWeight": 400,
        },
        "avatar": Object {
          "backgroundColor": "rgb(188, 188, 188)",
          "color": "#ffffff",
        },
        "badge": Object {
          "color": "#ffffff",
          "fontWeight": 500,
          "primaryColor": "#303F9F",
          "primaryTextColor": "#ffffff",
          "secondaryColor": "#C5CAE9",
          "secondaryTextColor": "#ffffff",
          "textColor": "#212121",
        },
        "baseTheme": Object {
          "fontFamily": "Roboto, sans-serif",
          "palette": Object {
            "accent1Color": "#C5CAE9",
            "accent2Color": "#C5CAE9",
            "accent3Color": "#C5CAE9",
            "alternateTextColor": "#ffffff",
            "borderColor": "#BDBDBD",
            "canvasColor": "#ffffff",
            "clockCircleColor": "rgba(0, 0, 0, 0.07)",
            "disabledColor": "#BDBDBD",
            "pickerHeaderColor": "#00bcd4",
            "primary1Color": "#303F9F",
            "primary2Color": "#3F51B5",
            "primary3Color": "#C5CAE9",
            "secondaryTextColor": " #757575",
            "shadowColor": "rgba(0, 0, 0, 1)",
            "textColor": "#212121",
          },
          "spacing": Object {
            "desktopDrawerMenuItemHeight": 48,
            "desktopDropDownMenuFontSize": 15,
            "desktopDropDownMenuItemHeight": 32,
            "desktopGutter": 24,
            "desktopGutterLess": 16,
            "desktopGutterMini": 8,
            "desktopGutterMore": 32,
            "desktopKeylineIncrement": 64,
            "desktopSubheaderHeight": 48,
            "desktopToolbarHeight": 56,
            "iconSize": 24,
          },
        },
        "borderRadius": 2,
        "bottomNavigation": Object {
          "backgroundColor": "#ffffff",
          "height": 56,
          "selectedColor": "#303F9F",
          "selectedFontSize": 14,
          "unselectedColor": "rgba(33, 33, 33, 0.54)",
          "unselectedFontSize": 12,
        },
        "button": Object {
          "height": 36,
          "iconButtonSize": 48,
          "minWidth": 88,
        },
        "card": Object {
          "fontWeight": 500,
          "subtitleColor": "rgba(33, 33, 33, 0.54)",
          "titleColor": "rgba(33, 33, 33, 0.87)",
        },
        "cardMedia": Object {
          "color": "rgba(255, 255, 255, 0.87)",
          "overlayContentBackground": "rgba(0, 0, 0, 0.54)",
          "subtitleColor": "rgba(255, 255, 255, 0.54)",
          "titleColor": "rgba(255, 255, 255, 0.87)",
        },
        "cardText": Object {
          "textColor": "#212121",
        },
        "checkbox": Object {
          "boxColor": "#212121",
          "checkedColor": "#303F9F",
          "disabledColor": "#BDBDBD",
          "labelColor": "#212121",
          "labelDisabledColor": "#BDBDBD",
          "requiredColor": "#303F9F",
        },
        "chip": Object {
          "backgroundColor": "rgb(224, 224, 224)",
          "deleteIconColor": "rgba(33, 33, 33, 0.26)",
          "fontSize": 14,
          "fontWeight": 400,
          "shadow": "0 1px 6px rgba(0, 0, 0, 0.12),
              0 1px 4px rgba(0, 0, 0, 0.12)",
          "textColor": "rgba(33, 33, 33, 0.87)",
        },
        "datePicker": Object {
          "calendarTextColor": "#212121",
          "calendarYearBackgroundColor": "#ffffff",
          "color": "#303F9F",
          "headerColor": "#00bcd4",
          "selectColor": "#3F51B5",
          "selectTextColor": "#ffffff",
          "textColor": "#ffffff",
        },
        "dialog": Object {
          "bodyColor": "rgba(33, 33, 33, 0.6)",
          "bodyFontSize": 16,
          "titleFontSize": 22,
        },
        "drawer": Object {
          "color": "#ffffff",
          "width": 256,
        },
        "dropDownMenu": Object {
          "accentColor": "#BDBDBD",
        },
        "enhancedButton": Object {
          "tapHighlightColor": "rgba(0, 0, 0, 0)",
        },
        "flatButton": Object {
          "buttonFilterColor": "#999999",
          "color": "rgba(0, 0, 0, 0)",
          "disabledTextColor": "rgba(33, 33, 33, 0.3)",
          "fontSize": 14,
          "fontWeight": 500,
          "primaryTextColor": "#303F9F",
          "secondaryTextColor": "#C5CAE9",
          "textColor": "#212121",
        },
        "floatingActionButton": Object {
          "buttonSize": 56,
          "color": "#303F9F",
          "disabledColor": "rgb(224, 224, 224)",
          "disabledTextColor": "#BDBDBD",
          "iconColor": "#ffffff",
          "miniSize": 40,
          "secondaryColor": "#C5CAE9",
          "secondaryIconColor": "#ffffff",
        },
        "fontFamily": "Roboto, sans-serif",
        "gridTile": Object {
          "textColor": "#ffffff",
        },
        "icon": Object {
          "backgroundColor": "#303F9F",
          "color": "#ffffff",
        },
        "inkBar": Object {
          "backgroundColor": "#C5CAE9",
        },
        "isRtl": false,
        "listItem": Object {
          "leftIconColor": "#757575",
          "nestedLevelDepth": 18,
          "rightIconColor": "#757575",
          "secondaryTextColor": " #757575",
        },
        "menu": Object {
          "backgroundColor": "#ffffff",
          "containerBackgroundColor": "#ffffff",
        },
        "menuItem": Object {
          "dataHeight": 32,
          "height": 48,
          "hoverColor": "rgba(33, 33, 33, 0.1)",
          "padding": 24,
          "rightIconDesktopFill": "#757575",
          "selectedTextColor": "#C5CAE9",
        },
        "menuSubheader": Object {
          "borderColor": "#BDBDBD",
          "padding": 24,
          "textColor": "#303F9F",
        },
        "overlay": Object {
          "backgroundColor": "rgba(0, 0, 0, 0.54)",
        },
        "palette": Object {
          "accent1Color": "#C5CAE9",
          "accent2Color": "#C5CAE9",
          "accent3Color": "#C5CAE9",
          "alternateTextColor": "#ffffff",
          "borderColor": "#BDBDBD",
          "canvasColor": "#ffffff",
          "clockCircleColor": "rgba(0, 0, 0, 0.07)",
          "disabledColor": "#BDBDBD",
          "pickerHeaderColor": "#00bcd4",
          "primary1Color": "#303F9F",
          "primary2Color": "#3F51B5",
          "primary3Color": "#C5CAE9",
          "secondaryTextColor": " #757575",
          "shadowColor": "rgba(0, 0, 0, 1)",
          "textColor": "#212121",
        },
        "paper": Object {
          "backgroundColor": "#ffffff",
          "color": "#212121",
          "zDepthShadows": Array [
            "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
            "0 3px 10px rgba(0, 0, 0, 0.16),
               0 3px 10px rgba(0, 0, 0, 0.23)",
            "0 10px 30px rgba(0, 0, 0, 0.19),
               0 6px 10px rgba(0, 0, 0, 0.23)",
            "0 14px 45px rgba(0, 0, 0, 0.25),
               0 10px 18px rgba(0, 0, 0, 0.22)",
            "0 19px 60px rgba(0, 0, 0, 0.3),
               0 15px 20px rgba(0, 0, 0, 0.22)",
          ],
        },
        "prepareStyles": [Function],
        "radioButton": Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#212121",
          "checkedColor": "#303F9F",
          "disabledColor": "#BDBDBD",
          "labelColor": "#212121",
          "labelDisabledColor": "#BDBDBD",
          "requiredColor": "#303F9F",
          "size": 24,
        },
        "raisedButton": Object {
          "color": "#ffffff",
          "disabledColor": "rgb(229, 229, 229)",
          "disabledTextColor": "rgba(33, 33, 33, 0.3)",
          "fontSize": 14,
          "fontWeight": 500,
          "primaryColor": "#303F9F",
          "primaryTextColor": "#ffffff",
          "secondaryColor": "#C5CAE9",
          "secondaryTextColor": "#ffffff",
          "textColor": "#212121",
        },
        "rawTheme": Object {
          "fontFamily": "Roboto, sans-serif",
          "palette": Object {
            "accent1Color": "#C5CAE9",
            "accent2Color": "#C5CAE9",
            "accent3Color": "#C5CAE9",
            "alternateTextColor": "#ffffff",
            "borderColor": "#BDBDBD",
            "canvasColor": "#ffffff",
            "clockCircleColor": "rgba(0, 0, 0, 0.07)",
            "disabledColor": "#BDBDBD",
            "pickerHeaderColor": "#00bcd4",
            "primary1Color": "#303F9F",
            "primary2Color": "#3F51B5",
            "primary3Color": "#C5CAE9",
            "secondaryTextColor": " #757575",
            "shadowColor": "rgba(0, 0, 0, 1)",
            "textColor": "#212121",
          },
          "spacing": Object {
            "desktopDrawerMenuItemHeight": 48,
            "desktopDropDownMenuFontSize": 15,
            "desktopDropDownMenuItemHeight": 32,
            "desktopGutter": 24,
            "desktopGutterLess": 16,
            "desktopGutterMini": 8,
            "desktopGutterMore": 32,
            "desktopKeylineIncrement": 64,
            "desktopSubheaderHeight": 48,
            "desktopToolbarHeight": 56,
            "iconSize": 24,
          },
        },
        "refreshIndicator": Object {
          "loadingStrokeColor": "#303F9F",
          "strokeColor": "#BDBDBD",
        },
        "ripple": Object {
          "color": "rgba(33, 33, 33, 0.87)",
        },
        "slider": Object {
          "handleColorZero": "#C5CAE9",
          "handleFillColor": "#ffffff",
          "handleSize": 12,
          "handleSizeActive": 18,
          "handleSizeDisabled": 8,
          "rippleColor": "#303F9F",
          "selectionColor": "#303F9F",
          "trackColor": "#C5CAE9",
          "trackColorSelected": "#C5CAE9",
          "trackSize": 2,
        },
        "snackbar": Object {
          "actionColor": "#C5CAE9",
          "backgroundColor": "#212121",
          "textColor": "#ffffff",
        },
        "spacing": Object {
          "desktopDrawerMenuItemHeight": 48,
          "desktopDropDownMenuFontSize": 15,
          "desktopDropDownMenuItemHeight": 32,
          "desktopGutter": 24,
          "desktopGutterLess": 16,
          "desktopGutterMini": 8,
          "desktopGutterMore": 32,
          "desktopKeylineIncrement": 64,
          "desktopSubheaderHeight": 48,
          "desktopToolbarHeight": 56,
          "iconSize": 24,
        },
        "stepper": Object {
          "backgroundColor": "transparent",
          "connectorLineColor": "#bdbdbd",
          "disabledTextColor": "rgba(0, 0, 0, 0.26)",
          "hoverBackgroundColor": "rgba(0, 0, 0, 0.06)",
          "hoveredIconColor": "#616161",
          "iconColor": "#303F9F",
          "inactiveIconColor": "#9e9e9e",
          "textColor": "rgba(0, 0, 0, 0.87)",
        },
        "subheader": Object {
          "color": "rgba(33, 33, 33, 0.54)",
          "fontWeight": 500,
        },
        "svgIcon": Object {
          "color": "#212121",
        },
        "table": Object {
          "backgroundColor": "#ffffff",
        },
        "tableFooter": Object {
          "borderColor": "#BDBDBD",
          "textColor": "#C5CAE9",
        },
        "tableHeader": Object {
          "borderColor": "#BDBDBD",
        },
        "tableHeaderColumn": Object {
          "height": 56,
          "spacing": 24,
          "textColor": "#C5CAE9",
        },
        "tableRow": Object {
          "borderColor": "#BDBDBD",
          "height": 48,
          "hoverColor": "#C5CAE9",
          "selectedColor": "#BDBDBD",
          "stripeColor": "rgba(151, 159, 207, 0.4)",
          "textColor": "#212121",
        },
        "tableRowColumn": Object {
          "height": 48,
          "spacing": 24,
        },
        "tabs": Object {
          "backgroundColor": "#303F9F",
          "selectedTextColor": "#ffffff",
          "textColor": "rgba(255, 255, 255, 0.7)",
        },
        "textField": Object {
          "backgroundColor": "transparent",
          "borderColor": "#BDBDBD",
          "disabledTextColor": "#BDBDBD",
          "errorColor": "#f44336",
          "floatingLabelColor": "#BDBDBD",
          "focusColor": "#303F9F",
          "hintColor": "#BDBDBD",
          "textColor": "#212121",
        },
        "timePicker": Object {
          "accentColor": "#303F9F",
          "clockCircleColor": "rgba(0, 0, 0, 0.07)",
          "clockColor": "#212121",
          "color": "#ffffff",
          "headerColor": "#00bcd4",
          "selectColor": "#3F51B5",
          "selectTextColor": "#ffffff",
          "textColor": "#ffffff",
        },
        "toggle": Object {
          "labelColor": "#212121",
          "labelDisabledColor": "#BDBDBD",
          "thumbDisabledColor": "#BDBDBD",
          "thumbOffColor": "#C5CAE9",
          "thumbOnColor": "#303F9F",
          "thumbRequiredColor": "#303F9F",
          "trackDisabledColor": "#C5CAE9",
          "trackOffColor": "#C5CAE9",
          "trackOnColor": "rgba(48, 63, 159, 0.5)",
          "trackRequiredColor": "rgba(48, 63, 159, 0.5)",
        },
        "toolbar": Object {
          "backgroundColor": "rgb(187, 191, 221)",
          "color": "rgba(33, 33, 33, 0.54)",
          "height": 56,
          "hoverColor": "rgba(33, 33, 33, 0.87)",
          "iconColor": "rgba(33, 33, 33, 0.4)",
          "menuHoverColor": "rgba(33, 33, 33, 0.1)",
          "separatorColor": "rgba(33, 33, 33, 0.175)",
          "titleFontSize": 20,
        },
        "tooltip": Object {
          "color": "#ffffff",
          "rippleBackgroundColor": "#616161",
        },
        "zIndex": Object {
          "appBar": 1100,
          "dialog": 1500,
          "dialogOverlay": 1400,
          "drawer": 1300,
          "drawerOverlay": 1200,
          "layer": 2000,
          "menu": 1000,
          "popover": 2100,
          "snackbar": 2900,
          "tooltip": 3000,
        },
      }
    }
  >
    <Component
      icon={
        <DeviceGpsFixed
          style={
            Object {
              "fill": "grey",
            }
          }
        />
      }
      title="Email"
      value="teste@example.com"
    >
      <div
        className="item"
      >
        <div
          className="itemIcon"
        >
          <DeviceGpsFixed
            style={
              Object {
                "fill": "grey",
              }
            }
          >
            <SvgIcon
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "fill": "grey",
                }
              }
              viewBox="0 0 24 24"
            >
              <svg
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "#212121",
                    "display": "inline-block",
                    "fill": "grey",
                    "height": 24,
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "width": 24,
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm8.94 3c-.46-4.17-3.77-7.48-7.94-7.94V1h-2v2.06C6.83 3.52 3.52 6.83 3.06 11H1v2h2.06c.46 4.17 3.77 7.48 7.94 7.94V23h2v-2.06c4.17-.46 7.48-3.77 7.94-7.94H23v-2h-2.06zM12 19c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
                />
              </svg>
            </SvgIcon>
          </DeviceGpsFixed>
        </div>
        <div
          className="itemText"
        >
          <a
            className="key"
          >
             
            Email
             
          </a>
          <a
            className="value"
          >
             
            teste@example.com
             
          </a>
        </div>
      </div>
    </Component>
  </MuiThemeProvider>
</Component>
`;
